### DISCUSSION TITLE
Diameter unit cm instead of m?

### DISCUSSION THREAD

**Discussion by lthurner** [e2nIEE/pandapipes#261] [Ideas]
The diameter unit is currently given in meters.

For normal pipe diameters it seems more user friendly to me to allow the input in cm instead of meters. "This pipe has a diameter of 86 cm" seems much more intuitive to me than "This pipe has a diameter of 0.86 m".

Is there a good reason to use meters instead of centimeters here?

**Comment by dlohmeier** [comment]
That is a good point. I remember that we once tried to mainly use basic SI units for the variables, but from the usability point of view, this doesn't really make much sense. Yet, this would also be a change to the API and requires some work to be backwards compatible, as we already learnt with pandapower v2.0 . If we agree to such a change, we should consider also changing other units, e.g. for mass flow from kg/s to kg/h, or rather use norm volume flows (e.g. for sinks), as they are typically less dependent on fluid properties (they can vary greatly between liquids and gases).

**Comment by jkisse** [comment]
Regarding the diameter, maybe millimeter would be a good choice, as it also used in the standards.

**Comment by lthurner** [comment]
Is there an equivalent function to the pandapower convert_format function in pandapipes?

**Comment by dlohmeier** [comment]
Yes, [convert_format](https://github.com/e2nIEE/pandapipes/blob/2e82196aed3a7d224594f219b971dd84d26b4c5d/pandapipes/io/convert_format.py#L18) also exists in pandapipes.

**Comment by SimonRubenDrauz** [comment]
I guess the best solution is the following: We change m to mm in case of diameter as this is the standard. In case of all other units: I would suggest we don't change them, however, we could think about convenience function which enable switches between the units.

**Comment by lthurner** [comment]
What about kg/s -> kg/h? I would not hesitate to change the units if it makes the data structure more user friendly. Better to change the units once than to sustain a suboptimal user interface...
